FordLabs/retroquest
# Migrate main header to react

## Overview
In this PR I've migrated the main header to react in the react version of the app. The contents of the pages are all still angular, but everything in the header, including the settings modal is now in react. 

Connects N/A

### Things that were done:
* Swapped out the angular version of the main header for a react version
* Added Cypress tests for the header
* Updated the settings modal to be able to change themes within the app and to log out
     * The theme will now be controlled by recoil app state and styles will be changed based on one root theme class vs classes being added everywhere 

* Moved Route and api paths to their own files so we have all paths in one place for the front end and then the back end
* Updated [React](https://reactjs.org/) to 17
* Added [Recoil](https://recoiljs.org/) to handle app state
* Removed the word "Beta" from the angular settings modal
* Converted react services from classes to objects
* Added in the "Info" tab that was recently added to the angular version and added the Retroquest logo in the new info tab (react version only)  
      * <img width="451" alt="Screen Shot 2022-01-25 at 5 09 50 PM" src="https://user-images.githubusercontent.com/74197928/151184754-3335f0c2-66a9-4f11-9620-c1a66b4421c4.png">
      * <img width="450" alt="Screen Shot 2022-01-25 at 5 09 37 PM" src="https://user-images.githubusercontent.com/74197928/151184775-5a8594eb-fdea-42db-9bdf-b486a05c92f9.png">

## Testing Instructions
1) Start the backend, and then start the front end in angular by running `start` and then run `start cypress` to ensure all cypress tests pass for the angular version of the app.
2) Run through using the angular app. Ensure everything works as before.
3) Start the backend, and then start the front end in react by running `start:react` and then run `start cypress` to ensure all cypress tests pass for the react version of the app.
4) Run through using the react app. Ensure everything works as before.
